### PR TITLE
RKE2 cluster by terraform

### PR DESCRIPTION
**Pull request by albinsun**
## Changes
1. Specify rancher provider version by `config.yml`:`terraform-provider-rancher`
   * Currently set `3.1.1` as default value due to a known issue https://github.com/rancher/terraform-provider-rancher2/issues/1284
   
2. Add TCs for RKE2 cluster using terraform rancher provider[^1], includes:
   * `rancher2_cloud_credential` resource
   * `rancher2_machine_config_v2` resource
   * `rancher2_cluster_v2` resource

## Verification
* `harvester-runtests/461` using `-k test_z_terraform`
  image

[^1]: https://github.com/rancher/terraform-provider-rancher2/tree/v3.1.1/docs/resources

### CODE REVIEW

File: harvester_e2e_tests/fixtures/terraform_rancher.py [hunk 1]
```diff
@@ -0,0 +1,349 @@
...
+    kubeconfig = res.text.replace("\\n", "\n").strip('"')
+
+    out, err, exc_code = \
+        tf_rancher.initial_provider(kubeconfig, tf_provider_version, harvester, rancher)
+    assert not err and 0 == exc_code
+    return tf_rancher
+
+
+@pytest.fixture(scope="session")
+def tf_resource(tf_provider_version):
```
Comment (lanfon72) [inline harvester_e2e_tests/fixtures/terraform_rancher.py:200]: we can use the exist one in terraform fixture.

File: harvester_e2e_tests/fixtures/terraform_rancher.py [hunk 2]
```diff
@@ -0,0 +1,349 @@
...
+    # Ref. https://registry.terraform.io/providers/rancher/rancher2/3.1.1/docs/resources/cluster_v2
+    url = f'{rancher["endpoint"]}/k8s/clusters/{harvester["id"]}/v1/harvester/kubeconfig'
+    auth = (rancher["access_key"], rancher["secret_key"])
+    data = {
+        "clusterRoleName": "harvesterhci.io:cloudprovider",
+        "namespace": "default",
+        "serviceAccountName": harvester["name"]
+    }
+    res = requests.post(url, auth=auth, json=data, verify=False)
+    kubeconfig = res.text.replace("\\n", "\n").strip('"')
```
Comment (lanfon72) [inline harvester_e2e_tests/fixtures/terraform_rancher.py:191]: the part to get kubeconfig from Rancher could move to `RancherAPI.generate_kubeconfig()` like HarvesterAPI did.

File: harvester_e2e_tests/fixtures/terraform_rancher.py (hunk 1 above)
Comment (albinsun) [reply harvester_e2e_tests/fixtures/terraform_rancher.py]: Currently using `tf_resource` and `tf_rancher_resource` which adopt different provider_version for respective suites.

**Comment by albinsun** [comment]
Fix comment requests and verified in `harvester-runtests#463`
image

**Comment by lanfon72** [comment]
@albinsun, I think you would need to rebase to reduce duplicated file changes from other (merged) PRs.
It would be something like:
```bash
# in local repo
git checkout main && git pull
git checkout <this-branch>
git rebase main
# then you will need to force push to update the PR
git push -f
```

**Comment by albinsun** [comment]
> @albinsun, I think you would need to rebase to reduce duplicated file changes from other (merged) PRs. It would be something like:
> 
> ```shell
> # in local repo
> git checkout main && git pull
> git checkout <this-branch>
> git rebase main
> # then you will need to force push to update the PR
> git push -f
> ```

@lanfon72, thanks for ur comment, will use rebase instead of merge then.

**Review by khushboo-rancher — APPROVED** [review]

**Review by lanfon72 — APPROVED** [review]

**Pull request merged**